core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdbutil.go

```diff
@@ -134,19 +146,50 @@ func createCouchDatabase(couchInstance *couchInstance, dbName string) (*couchDat
 	return &couchDBDatabase, nil
 }
 
+// legacy versions of Couch prior to 3.0.0 did not require a userID and password to be configured when starting
+// the database. If the _users table did not exist in Couch < 3.0 the peer startup process created the _users
+// table. In 3.0.0 the _users table cannot be modified directly. This function can be used to determine if legacy (< 3.0.0)
+// operations can and should be performed.
+func checkLegacyCouchDBVersion(couchVersion string) (bool, error) {
+	couchSemVersion, err := version.NewSemver(couchVersion)
+	if err != nil {
+		return false, errors.Wrap(err, "error parsing CouchDB semver")
+	}
+	gateSemVersion, err := version.NewSemver("3.0.0")
+	if err != nil {
+		return false, errors.Wrap(err, "error parsing default CouchDB semver")
+	}
+	return couchSemVersion.LessThan(gateSemVersion), nil
```

Comment:
Do we need hash-graph's version matching pkg? I thought we would always support one CouchDB version. At the worst case, we might support the previous and current version. Can't we simply compare strings (`"3.1.0", "2.3"`) rather than using a new dependency and making this code generic? Further, LessThan might return true even if the deployment uses `2.0` CouchDB. Hence, I am confused a bit. If there is a plan to support multiple CouchDB versions (more than 2) officially, it would create complication for us whenever we touch `statecouchdb` pkg.

Reply:
With this we would support 3.1.x and 2.3.x. Today in fabric we actually test against 2.3 (though there is no actual 2.3, its whatever the latest 2.3.x version is), so it's whatever is latest, no idea until runtime what the patch level is. And I actually only chose to switch to the semver package because it seemed less error prone than doing string splits and comparisons. I also only elected to use it as the package is already part of fabrics dep tree.

If there is other dissent on this I'll switch it to a string matcher